lib/spack/spack/cmd/export.py

```diff
@@ -0,0 +1,196 @@
+        msg = msg.format(' '.join(args.constraint))
+        tty.msg(msg)
+        return
+
+    packages = spack.config.get('packages', scope=args.scope)
+
+    # If tags have been specified on the command line, filter by tags
+    if args.tags:
+        packages_with_tags = spack.repo.path.packages_with_tags(*args.tags)
+        specs = [x for x in specs if x.name in packages_with_tags]
+
+    cls = None
+    if args.module:
+        cls = spack.modules.module_types[args.module]
+
+    # Collect packages per package name
+    pkgs = {}
+    for spec in specs:
+        pkgs.setdefault(spec.name, []).append(spec)
+
+    pymods = {}
+
+    # Dump per package, make sure that none are forgotten
+    for pkg, pkg_specs in pkgs.items():
+        paths = syaml_dict()
+        modules = syaml_dict()
+        package = packages.setdefault(pkg, syaml_dict())
+        versions = None
+        if 'version' in package:
+            versions = [str(v) for v in package['version']]
+
+        for spec in pkg_specs:
+            key = spec.format(args.format)
+            sflags = []
+            bflags = []
+            for k, v in spec.variants.items():
+                default = None
+                if k in spec.package.variants:
+                    default = spec.package.variants[k].default
+                if v.value != default:
+                    if v.value in (True, False):
+                        bflags.append(v)
+                    elif v.name != 'patches':
+                        sflags.append(v)
+
+            sflags = ' '.join(str(f) for f in sorted(sflags))
+            bflags = ''.join(str(f) for f in sorted(bflags))
+            key = ' '.join([e for e in (key, sflags, bflags) if len(e) > 0])
+
+            if isinstance(spec.package, PythonPackage):
+                py = spec['python']
+                if args.dependencies:
+                    key += " ^{0}".format(py.format("$_$@"))
+
+                if not spec.package.is_activated(py.package.view()):
+                    # For external packages, setup_environment is not
+                    # called, and thus they are not included in
+                    # PYTHON_PATH.
+                    msg = "python package not activated, skipping: {0}"
+                    msg = msg.format(spec.format("$_$@"))
+                    tty.warn(msg)
+                    # paths[key] = str(spec.prefix)
+                else:
+                    mod = pymods.setdefault(py, cls(py) if cls else None)
+                    if mod and not mod.conf.blacklisted:
+                        if os.path.exists(mod.layout.filename):
+                            paths[key] = '/activated'
+                            # modules[key] = mod.layout.use_name
+                            # paths[key] = str(spec.prefix)
+                    else:
+                        msg = "python package activated in inactive module, skipping: {0}"
+                        msg = msg.format(spec.format("$_$@"))
```

Comment:
this one I have to understood better.  I thought if package is activate and module is generated (not blacklisted) then it should be exported. But I see:

```
 → spack activate py-six ^python@3:~dbm
==> Package py-six/e6nzcw5 is already activated.

→ spack module tcl refresh --delete-tree -y
==> Regenerating tcl module files
==> Warning: Could not write module file [/Users/kumbhar/workarena/software/sources/spack/share/spack/modules/darwin-sierra-x86_64/coreneuron/ring-clang]
==> Warning: 	--> Package neuronmodelresource not found. <--

 → module show py-six
-------------------------------------------------------------------
/Users/kumbhar/workarena/software/sources/spack/share/spack/modules/darwin-sierra-x86_64/py-six/1.11.0-clang:

module-whatis	Python 2 and 3 compatibility utilities.
conflict	py-six
prepend-path	DYLD_LIBRARY_PATH	/Users/kumbhar/workarena/software/sources/spack/opt/spack/darwin-sierra-x86_64/clang-9.0.0-apple/py-six-1.11.0-vyvpf7kuslhbc7fy4gkslzgajaa55csq/lib
prepend-path	DYLD_FALLBACK_LIBRARY_PATH	/Users/kumbhar/workarena/software/sources/spack/opt/spack/darwin-sierra-x86_64/clang-9.0.0-apple/py-six-1.11.0-vyvpf7kuslhbc7fy4gkslzgajaa55csq/lib
prepend-path	CMAKE_PREFIX_PATH	/Users/kumbhar/workarena/software/sources/spack/opt/spack/darwin-sierra-x86_64/clang-9.0.0-apple/py-six-1.11.0-vyvpf7kuslhbc7fy4gkslzgajaa55csq/
prepend-path	PYTHONPATH	/Users/kumbhar/workarena/software/sources/spack/opt/spack/darwin-sierra-x86_64/clang-9.0.0-apple/py-six-1.11.0-vyvpf7kuslhbc7fy4gkslzgajaa55csq/lib/python3.6/site-packages
prepend-path	PYTHONPATH	/Users/kumbhar/workarena/software/sources/spack/opt/spack/darwin-sierra-x86_64/clang-9.0.0-apple/py-sh-1.12.9-2x6rjxs6k4i65tqlxytac4h2mg4hn3sv/lib/python3.6/site-packages
setenv		PY_SIX_ROOT	/Users/kumbhar/workarena/software/sources/spack/opt/spack/darwin-sierra-x86_64/clang-9.0.0-apple/py-six-1.11.0-vyvpf7kuslhbc7fy4gkslzgajaa55csq
-------------------------------------------------------------------

 → spack export --scope user py-six
==> Warning: python package activated in inactive module, skipping: py-six@1.11.0
packages: {}
```

Reply:
I put that in to avoid listing python modules that are not activated in the parent python package. I'll tune that.

Reply:
I will actually keep skipping these. Modules for python packages have the same problem with versioning/dependencies as regular extensions: How do we tell spack for which version a module is?